torch_xla/utils/checkpoint.py

```diff
@@ -143,17 +144,21 @@ def backward(ctx, *args):
       rng_devices = ctx.fwd_gpu_devices
     xm.optimization_barrier_(
         CheckpointFunction._extract_tensors_from_list(inputs + list(args)))
+    # torch.random.fork_rng will handle the cpu and gpu seed
+    # xm.fork_rng will handle the xla device seed
     with torch.random.fork_rng(
         devices=rng_devices, enabled=ctx.preserve_rng_state):
-      if ctx.preserve_rng_state:
-        torch.set_rng_state(ctx.fwd_cpu_state)
-        if ctx.had_cuda_in_fwd:
-          set_device_states(ctx.fwd_gpu_devices, ctx.fwd_gpu_states)
-      detached_inputs = detach_variable(tuple(inputs))
-      with torch.enable_grad(), \
-           torch.cuda.amp.autocast(**ctx.gpu_autocast_kwargs), \
-           torch.cpu.amp.autocast(**ctx.cpu_autocast_kwargs):
-        outputs = ctx.run_function(*detached_inputs)
+      with xm.fork_rng():
```

Comment:
any reason not to pass the `rng_devices ` and `ctx.preserve_rng_state` ?

Reply:
It looks like the upstream code doesn't reset the state. @YangFei1990 Do you know why?

Reply:
I guess upstream seed is handled by `torch.random.fork_rng`? through I am not sure why it doesn't work with pytorch/xla...

Reply:
Yeah upstream seed is handled by `torch.random.fork_rng`. It will fork torch seed but somehow it won't set XLA's RNG. This seed `torch_xla._XLAC._xla_get_rng_seed(str(device)` is it independent to torch seed? How torch XLA in general handle RNGs?

Reply:
I did not change the previous behavior, i.e. upstream seed will still be maintained as it was (check code below). I simply add another preserve RNG states.